src/navigators/RootNavigator.tsx

```diff
@@ -1,95 +1,71 @@
 import { ROOT_NAVIGATOR_ROUTES, RootStackParamList } from "config/routes";
+import { AUTH_STATUS } from "config/types";
 import { useAuthenticationStore } from "ducks/auth";
-import { isHashKeyValid } from "hooks/useGetActiveAccount";
 import { AuthNavigator } from "navigators/AuthNavigator";
 import { TabNavigator } from "navigators/TabNavigator";
-import React, { useEffect, useState } from "react";
+import React, { useEffect, useMemo, useState } from "react";
 import RNBootSplash from "react-native-bootsplash";
-import { dataStorage } from "services/storage/storageFactory";
 
 const RootStack = createNativeStackNavigator<RootStackParamList>();
 
 export const RootNavigator = () => {
-  const { getIsAuthenticated, isAuthenticated } = useAuthenticationStore();
-  const [hasAccount, setHasAccount] = useState<boolean | null>(null);
+  const { authStatus, getAuthStatus } = useAuthenticationStore();
   const [initializing, setInitializing] = useState(true);
 
   useEffect(() => {
-    const validateAuth = async () => {
-      try {
-        // First check if hash key is valid - this might authenticate the user
-        const hashKeyValid = await isHashKeyValid();
-        if (!hashKeyValid) {
-          setInitializing(false);
-          return;
-        }
-
-        // Then get auth status
-        getIsAuthenticated();
-
-        // Check if we have a stored account
-        const activeAccountId = await dataStorage.getItem(
-          STORAGE_KEYS.ACTIVE_ACCOUNT_ID,
-        );
-        setHasAccount(!!activeAccountId);
-      } catch (error) {
-        logger.error(
-          "RootNavigator.validateAuth",
-          "Error initializing the app",
-          error,
-        );
-      } finally {
-        await RNBootSplash.hide({ fade: true });
-        setInitializing(false);
-      }
+    const initializeApp = async () => {
+      // Check auth status when app starts
+      await getAuthStatus();
+      setInitializing(false);
+      RNBootSplash.hide({ fade: true });
     };
 
-    validateAuth();
-  }, [getIsAuthenticated]);
-
-  // Show loading screen while initializing
-  if (initializing) {
-    return null;
-  }
+    initializeApp();
+  }, [getAuthStatus]);
 
-  // Determine initial route
-  const getInitialRouteName = () => {
-    if (isAuthenticated) {
-      return ROOT_NAVIGATOR_ROUTES.MAIN_TAB_STACK;
+  // Make the stack re-render when auth status changes
+  const initialRouteName = useMemo(() => {
+    if (authStatus === AUTH_STATUS.NOT_AUTHENTICATED) {
+      return ROOT_NAVIGATOR_ROUTES.AUTH_STACK;
     }
 
-    // If we have an account but not authenticated, show login screen
-    if (hasAccount) {
-      return ROOT_NAVIGATOR_ROUTES.AUTH_STACK;
+    if (authStatus === AUTH_STATUS.HASH_KEY_EXPIRED) {
+      return ROOT_NAVIGATOR_ROUTES.LOCK_SCREEN;
     }
 
-    // Otherwise show the auth stack (welcome, signup, etc.)
-    return ROOT_NAVIGATOR_ROUTES.AUTH_STACK;
-  };
+    return ROOT_NAVIGATOR_ROUTES.MAIN_TAB_STACK;
```

Comment:
Should we consider `AUTH_STATUS.NOT_AUTHENTICATED` as default instead of `AUTH_STATUS.AUTHENTICATED` ?

The way it's implemented now works but I think it's a _little_ scary to consider "authenticated" as the default behavior here. Wdyt?